tests/Microsoft.Experimental.Collections.Tests/Microsoft/Collections/Extensions/OrderedDictionaryTests.cs

```diff
@@ -1475,6 +1475,35 @@ public void RemoveSlotReused()
             Assert.Equal(3, d.Count);
         }
 
+        [Fact]
+        public void Insert_Succeeds_WhenThereAreCollisions()
```

Comment:
Might be worth test variations for 
- Removing start of chain vs. within chain
- Insert at the end
- Remove at the end

- Remove at the end, in both cases where the entry is part of a chain. Ideally, where it